apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/DynamicTransformer.java

```diff
@@ -38,23 +48,13 @@
      * @param classToInstrument      the class which should be instrumented.
      * @param instrumentationClasses the instrumentation which should be applied to the class to instrument.
      */
-    void ensureInstrumented(Class<?> classToInstrument, Collection<Class<? extends ElasticApmInstrumentation>> instrumentationClasses);
-
-    class Accessor {
-        private static final DynamicTransformer transformer;
-
-        static {
-            ClassLoader classLoader = Accessor.class.getClassLoader();
-            if (classLoader == null) {
-                classLoader = ClassLoader.getSystemClassLoader();
-            }
-            // loads the implementation provided by the core module without depending on the class or class name
-            transformer = ServiceLoader.load(DynamicTransformer.class, classLoader).iterator().next();
-        }
-
-        public static DynamicTransformer get() {
-            return transformer;
-        }
+    public static void ensureInstrumented(Class<?> classToInstrument, Collection<Class<? extends ElasticApmInstrumentation>> instrumentationClasses) {
+        transformer.ensureInstrumented(classToInstrument, instrumentationClasses);
     }
 
+    public interface DynamicTransformerSupplier {
```

Comment:
[minor] I don't think `DynamicTransformerSupplier` is a good name for this interface. I would call it `DynamicTransformer` and find a better name for the declaring class, even `StaticDynamicTransformer`, but preferably something better 🙂

Reply:
I agree that the name of the interface isn't ideal. But I wanted to optimize for the actual user facing class which is `DynamicTransformer`. While `DynamicTransformerSupplier` needs to be public, users never really get to see it. Also, it's in line with `WeakConcurrentSupplier`.

Reply:
It's just that `DynamicTransformer` doesn't supply anything (as opposed to the weak map supplier).
What about `RuntimeInstrumentor` for the interface?

And an additional suggestion - I just noticed that `DynamicTransformerImpl` seems redundant, why not make `ElasticApmAgent` the service provider implementing the interface (whatever it's called 🙂  ).

Either way, feel free to ignore both suggestions and merge if you disagree. It's not that important.